run_analysis/run_analysis.R

```diff
@@ -222,13 +223,38 @@ run_analysis <- function(data, settings) {
 
   if (length(errors$errors) == 0) {
     logging::logdebug("error checks passed", logger=logger)
-
+    
+    # First get ideal response profile for 2015 standard, AS4777.2:2015.
     response_data <- ideal_response_from_frequency(
-      data$frequency_data, settings$region_to_load
+      data$frequency_data, settings$region_to_load, f_ulco=50.25, f_hyst=0.1, t_hyst=60, f_upper=52.00
     )
     data$ideal_response_to_plot <- response_data$ideal_response_to_plot
     data$region_frequency <- response_data$region_frequency
-
+    
+    # Next, get ideal response profile for 2020 standard, AS4777.2:2020 (uses different settings based on region).
+    # Currently, WA (Western Power) uses "Australia B", TAS uses "Australia C", all other NEM regions use "Australia A".
+    if(settings$region_to_load == "WA"){
+      f_ulco <- 50.15
+      f_hyst <- 0.1
+      t_hyst <- 20
+      f_upper <- 52.00
+    } else if(settings$region_to_load == "TAS"){
+      f_ulco <- 50.5
+      f_hyst <- 0.05
+      t_hyst <- 20
+      f_upper <- 55.00
+    } else{
+      f_ulco <- 50.25
+      f_hyst <- 0.1
+      t_hyst <- 20
+      f_upper <- 52.00
+    }
```

Comment:
This if/else if may be best as a function if you think we might use it elsewhere at any stage